src/MongoDBRepository.php

```diff
@@ -63,6 +63,10 @@ public function find($id)
      */
     public function findBy(array $fields): array
     {
+        if (empty($fields)) {
```

Comment:
I think it makes more sense if we throw an Exception in this case, as requesting to filter on 0 fields seems like an error.
Alternatively it could mean 'no filter', but in that case `findAll` should have been called.

What do you think?

Reply:
I don't know, in this way we change the contract of this class. Everyone uses the `MongoDBRepository` now knows that if he calls `findBy` with an empty array should not receive an exception. For example, Doctrine repository doesn't throw an exception if you call `findBy` with an empty array and `findAll` is a simple alias for `findBy([])`. Another case could be if you generate the filter fields array dynamically and you want to have the possibility to pass an empty array when nobody needs a filter instead to put an `if` for selecting a different method based on array filter.

Reply:
I think we should just accept this as it is the same as the Elasticsearch read model repository: https://github.com/broadway/read-model-elasticsearch/blob/master/src/ElasticSearchRepository.php#L91-L103
Which I think should have been the correct implementation to start with.